spot-contracts/contracts/strategies/FeeStrategy.sol

```diff
@@ -0,0 +1,226 @@
+    /// @param maxBurnFeePerc_ The new burn fee ceiling percentage
+    ///        as a fixed point number with {PERC_DECIMALS} places.
+    function updateBurnFees(uint256 maxBurnFeePerc_) external onlyOwner {
+        require(maxBurnFeePerc_ <= HUNDRED_PERC, "FeeStrategy: burn fee too high");
+        maxBurnFeePerc = maxBurnFeePerc_;
+    }
+
+    /// @notice Update the parameters determining the slope and asymptotes of the sigmoid fee curve.
+    /// @param p Lower, Upper and Growth sigmoid paramters are fixed point numbers with {PERC_DECIMALS} places.
+    function updateRolloverFees(SigmoidParams memory p) external onlyOwner {
+        require(p.lower >= -int256(MAX_PERC), "FeeStrategy: fee bound too low");
+        require(p.upper <= int256(MAX_PERC), "FeeStrategy: fee bound too high");
+        rolloverFeeAPR.lower = p.lower;
+        rolloverFeeAPR.upper = p.upper;
+        rolloverFeeAPR.growth = p.growth;
+    }
+
+    /// @inheritdoc IFeeStrategy
+    function computeMintFeePerc() external override returns (uint256) {
+        // when vault tvl > target, we encorage supply growth by dropping mint fees to 0
+        return (computeDeviationRatio(perp.getDepositBond()) > HUNDRED_PERC) ? 0 : maxMintFeePerc;
+    }
+
+    /// @inheritdoc IFeeStrategy
+    function computeBurnFeePerc() external override returns (uint256) {
+        // when vault tvl < target, we encorage supply reduction by dropping burn fees to 0
+        return (computeDeviationRatio(perp.getDepositBond()) < HUNDRED_PERC) ? 0 : maxBurnFeePerc;
+    }
+
+    /// @inheritdoc IFeeStrategy
+    function computeRolloverFeePerc() external override returns (int256) {
+        IBondController referenceBond = perp.getDepositBond();
+        int256 rolloverAPR = Sigmoid.compute(
+            computeDeviationRatio(referenceBond).toInt256(),
+            rolloverFeeAPR.lower,
+            rolloverFeeAPR.upper,
+            rolloverFeeAPR.growth,
+            PERC_DECIMALS
+        );
+
+        // We calculate the rollover fee for the given cycle by dividing the annualized rate
+        // by the number of cycles in any given year.
+        return (rolloverAPR * referenceBond.duration().toInt256()) / ONE_YEAR_SEC;
+    }
+
+    /// @inheritdoc IFeeStrategy
+    function decimals() external pure override returns (uint8) {
+        return PERC_DECIMALS;
+    }
+
+    /// @notice Computes the ratio between the vault's TVL and the target TVL.
+    /// @dev Adjusts the computed result based on the `deviationTresholdPerc`.
+    /// @return The ratio as a fixed point number with {PERC_DECIMALS} places.
+    function computeDeviationRatio(IBondController referenceBond) public returns (uint256) {
+        // NOTE: Ensure that the perp's TVL and vault's TVL have the same base denomination.
+        (uint256 perpRatio, ) = perp.computeEffectiveTrancheRatio(referenceBond);
+        uint256 targetTVL = perp.getTVL().mulDiv(TRANCHE_RATIO_GRANULARITY, perpRatio);
```

Comment:
so we're committing to needing 4x rather than 3x in the vault for 25:75 bonds?

Reply:
For now that is the case. Once we allow for flash-rollovers, we need to revisit this